packages/cli/src/cli/basemaps/tileset.config.ts

```diff
@@ -0,0 +1,59 @@
+import * as z from 'zod';
+
+const zImageryConfig = z.object({
+    imgId: z.string(),
+    name: z.string(),
+    priority: z.number().refine((val) => val >= -1 && val <= 10000, { message: 'must be between -1 and 10000' }),
+    zoom: z
+        .array(z.number())
+        .min(2)
+        .max(2)
+        .refine(([min, max]) => min >= 0 && max <= 32 && min <= max, {
+            message: 'range must be within [0, 32]',
+        }),
+    production: z.boolean().optional(),
+});
+
+/**
+ * The projections supported by TileSetConfig
+ */
+export enum ProjectionCode {
+    Google = 'EPSG:3857',
+    Nztm2000 = 'EPSG:2193',
+}
+
+const zBackground = z
+    .string()
+    .refine((val) => /^[0-9a-f]{8}$/.test(val), { message: 'must have 8 lower case hex characters' });
```

Comment:
would be safer to just parse this and make sure it actually works

Reply:
The parse function is super lax so not safer

Reply:
Safer as in it actually matches what the parser does, this already would fail on things that use to work eg `0x000000` or `#FF00FF`

Reply:
The parse function accepts `'1anyting'` I'll make the parse function more robust